pylint/checkers/base/name_checker/checker.py

```diff
@@ -410,25 +411,43 @@ def visit_assignname(self, node: nodes.AssignName) -> None:
             if isinstance(assign_type, nodes.Assign):
                 inferred_assign_type = utils.safe_infer(assign_type.value)
 
-                # Check TypeVar's assigned alone or in tuple assignment
-                if isinstance(node.parent, nodes.Assign) and self._assigns_typevar(
-                    assign_type.value
-                ):
-                    self._check_name("typevar", assign_type.targets[0].name, node)
-                elif (
+                # Check TypeVar's and TypeAliases assigned alone or in tuple assignment
+                if isinstance(node.parent, nodes.Assign):
+                    if self._assigns_typevar(assign_type.value):
+                        self._check_name("typevar", assign_type.targets[0].name, node)
+                        return
+                    if self._assigns_typealias(assign_type.value):
+                        self._check_name("typealias", assign_type.targets[0].name, node)
+                        return
+
+                if (
                     isinstance(node.parent, nodes.Tuple)
                     and isinstance(assign_type.value, nodes.Tuple)
                     # protect against unbalanced tuple unpacking
                     and node.parent.elts.index(node) < len(assign_type.value.elts)
-                    and self._assigns_typevar(
-                        assign_type.value.elts[node.parent.elts.index(node)]
-                    )
                 ):
-                    self._check_name(
-                        "typevar",
-                        assign_type.targets[0].elts[node.parent.elts.index(node)].name,
-                        node,
-                    )
+                    if self._assigns_typevar(
+                        assign_type.value.elts[node.parent.elts.index(node)]
```

Comment:
`assign_type.value.elts[node.parent.elts.index(node)]` is used for both method calls (`_assigns_typevar`, `_assigns_typealias`). Maybe it would make sense to create a temp var for it.

Thinking one step further, the following is also reused.
```py
                            assign_type.targets[0]
                            .elts[node.parent.elts.index(node)]
                            .name
```
Would it make sense to combine both methods (`_assigns_typevar` and `_assigns_typealias`) and simply pass it `assign_type`, `node`, and a "mode" argument?

Reply:
I made a new temp var.

Merging doesn't make too much sense I think. The inference is different `utils.safe_infer(node.func)` and `utils.safe_infer(node)` and we check for different `qnames`. Based on the `assign_type` we would simply follow completely different code paths.